internal/pkg/composable/providers/kubernetes/pod.go

```diff
@@ -437,8 +437,14 @@ func generateContainerData(
 
 				if config.Hints.Enabled { // This is "hints based autodiscovery flow"
 					if !managed {
-						hintsMapping := getHintsMapping(k8sMapping, logger, config.Prefix, c.ID)
+						hintsMapping, processorMapping := getHintsMapping(k8sMapping, logger, config.Prefix, c.ID)
```

Comment:
It's better if you avoid returning 2 different values from a single function. I would suggest:
1) Either define a standalone `getProcessors` to only return the processors.
2) Or return a struct with `hintsMapping` and `processors` as fields.

Based on the fact that the returned processors are not conceptually a mapping like the `hintsMapping` I would go for the 1st option. This would provide better isolation and would be more testable with unit-tests.
Let me know what you think :).

Reply:
> For `Either define a standalone getProcessors to only return the processors.`

The reason that I chose this way and return two values is that `getHintsMapping` function already calculates the hints (https://github.com/elastic/elastic-agent/pull/3107/files#diff-ed933ad9cf09eb9ca797c5c89b5e8991624cbe5301d88f996d9bd947f4e3ea16R512) and I base my extra calculations for extracting of processors on the hints extracted. If I implement it on another function this means that I will have two call twice the GenerateHintsMapping and this imposes extra overhead

> For struct

This can be an alternative but still having second thoughts for a 2-value struct. This would also mean changes to other places and tests.

Reply:
👍🏽 for avoiding the extra call.
The struct though would make it more readable and beautiful code-wise.

Reply:
Added the struct see https://github.com/elastic/elastic-agent/pull/3107/files#diff-ed933ad9cf09eb9ca797c5c89b5e8991624cbe5301d88f996d9bd947f4e3ea16R487